### PR TITLE
Update cpack lecture

### LINKED ISSUE #63: Optimize CPack/Debian lecture
In WT 2021/22, the CMake lecture contained information on CPack, but not on "make install", even though we had no time to do the CPack things in the end. It might be a better fit to do the `make install` still in the CMake lecture and only introduce CPack in the "CPack/Debian" lecture.

Moreover, the CPack/Debian lecture started rather theoretically. An idea to improve this could be start with the "Hello World" example code from the CMake lecture again. Then, quickly repeat what we already know. Then, motivate a long the lines 

> We already know how to package Python code ... This was very helpful to give our code to other ... But how can we package C++ code?

### PR DESCRIPTION
**Pull request by uekerman**
## Description

Update CPack lecture:

- Rewrote and re-organized demo notes
- Moved example code from [separate repo](https://github.com/Simulation-Software-Engineering/HelloWorld) to `examples/cpack`
- Added a misc part "more cmake demo" to show preCICE things and ccmake

Fixes #63

## Checklist

- [x] I made sure that the markdown files are formatted properly.
- [ ] I made sure that all hyperlinks are working.

- [x] I added the new content to the `README.md` inside the topics subdirectory, e.g., `01_version_control/README.md`.
- [x] I added the new content to the `timetable.md` in the root of this repository.

### CODE REVIEW

File: 03_building_and_packaging/cpack_demo.md [hunk 1]
```diff
@@ -1,63 +1,28 @@
...
+- Build and start Docker container:
+    - `docker build -t "cpack_demo"`
+    - `docker run -it cpack_demo`
+
+## Add Install Target to CMake Configuration
+
+- Show `CMakeLists.txt`
+- First observations:
+    - We still build a library (`add_library`) and then an executable (`add_executable`).
+    - We do this **for demonstration only**. There is normally no need handle internal dependencies that complicated.
```
Comment (IshaanDesai) [inline 03_building_and_packaging/cpack_demo.md:16]:
```suggestion
    - We do this **for demonstration only**. There is normally no need handle internal dependencies in such a complicated way.
```

File: 03_building_and_packaging/cpack_demo.md [hunk 2]
```diff
@@ -1,63 +1,28 @@
...
+    - Build the code: We still get an executable `helloworld` and a library `libsse.a`.
+- Changes compared to previous version / last week:
+    - Version number, not needed yet, but good practice
+    - `set_target_properties`: Mark include files needed for `libsse` library (`sse/sse.hpp`) as publicly visible.
+        - Needed such that CMake knows which headers should be public and should be installed. By default CMake assumes headers are for internal use (within the current project) only.
+    - `target_include_directories`: Where do we have to look for the dependency depending on the type of build (internally, top-level cmake project, external dependency) TODO
+    - Include CMake macro `GNUInstallDirs` to get predefined variables such as installation directories. Directories names and their purpose are defined in the [GNU coding standard](https://www.gnu.org/prep/standards/html_node/Directory-Variables.html). TODO
```
Comment (IshaanDesai) [inline 03_building_and_packaging/cpack_demo.md:23]: Same as above.

**Review by IshaanDesai — APPROVED** [review]
Just a few minor comments. I did not go through each and every line but just had a high level look. Content looks good :+1:

File: 03_building_and_packaging/cpack_demo.md [hunk 3]
```diff
@@ -1,63 +1,28 @@
...
+- Show `CMakeLists.txt`
+- First observations:
+    - We still build a library (`add_library`) and then an executable (`add_executable`).
+    - We do this **for demonstration only**. There is normally no need handle internal dependencies that complicated.
+    - Build the code: We still get an executable `helloworld` and a library `libsse.a`.
+- Changes compared to previous version / last week:
+    - Version number, not needed yet, but good practice
+    - `set_target_properties`: Mark include files needed for `libsse` library (`sse/sse.hpp`) as publicly visible.
+        - Needed such that CMake knows which headers should be public and should be installed. By default CMake assumes headers are for internal use (within the current project) only.
+    - `target_include_directories`: Where do we have to look for the dependency depending on the type of build (internally, top-level cmake project, external dependency) TODO
```
[the remark this answers is not in the document]
Comment (uekerman) [reply 03_building_and_packaging/cpack_demo.md]: No, good catch!

**Pull request merged**